docs/DeveloperGuide.md

```diff
@@ -1,34 +1,422 @@
+
+root :
+lsc -m cs3235
+Here are what you are looking for...
+
++--------------------------------------------------------------------------------------------------+
+ NO |     MODULE     |                                CATEGORY                                | PTY
++--------------------------------------------------------------------------------------------------+
+ 1  |     CS3235     |                               Assignment                               |  4
+ 2  |     CS3235     |                                  Lab                                   |  3
+ 3  |     CS3235     |                                Lecture                                 |  1
+ 4  |     CS3235     |                                Tutorial                                |  2
++--------------------------------------------------------------------------------------------------+
+Total categories: 5
++--------------------------------------------------------------------------------------------------+
+```
+
+1. James has two choices:
+
+   1. enter `cd cs3235` to enter the module directory then enter `addc misc` to add the category.
+   2. enter `addc misc -m cs3235` to add the category at the root directory.
+
+   Suppose James use the first method, after the second input, the input is parsed as an **add task** command and executed, the `AddCategoryCommand#execute()` will call ``AddCategoryCommand#getParentDirectory()` to get the current parent directory, then it will instantiate an `Category` object with the name "misc". After which `CategoryManager#add()` will be called to add the new object. In the `CategoryManager#add()` method, it will call `CategoryManager#contains()` method to check if the current parent directory contains the category with name "misc", finally add the object into the `ArrayList` of `categoryList`.
+
+2. James receive the following feedback:
+
+   ```
+   root :
+   addc misc
+   SUCCESS!! Category misc is created.
+   ```
+
+
+
+Below is a *sequence diagram* to illustrate the above example scenario.  
+
+```
+to-do: add the sequence diagram
+```
+
+
+
+### 2. Delete Feature  
+#### Overview  
+The **delete** feature deletes *modules*, *categories* and *tasks* from the Module, Category and Task List respectively.   
+When the user first requests to execute the **delete** command to delete a directory by providing its name, the application will first filter for directories with matching names. From here, there are **three** possible outcomes:  
+
+1. There are **no** matches --  Nothing is deleted.
+2. There is **one** match -- A prompt will be given to the user to confirm the deletion.
+3. There are **multiple** matches -- The list of matches will be shown to the user, and the user chooses which ones to delete. A further prompt will be given to confirm the deletion(s).
+
+#### Feature Implementation  
+![Delete Command Class Diagram](images/Delete_Command_Class_Diagram.png)
```

Comment:
I think in my opinion, there are too many detail in one diagram. I suggest we can keep it simple how it is shown in Add :)

Reply:
I agree. maybe it could be cleaned up by removing unimportant attributes in the leaf classes